### PR TITLE
feat/json-ld_with_did_resolver

### PR DESCRIPTION
**Pull request by burdettadam**
This PR updates json-ld admin routes to utilize did resolver during verification. Before, a user-provided a verkey to verify json-ld signatures, now if a verkey is not provided the `verificationMethod` will be used to attempt to resolve the verkey. Code updates to support this feature include:
- dropped attribute reporting
- interface update, pass session instead of wallet instance
- improved tests
- PyDID exception wrapping

This PR is intended not to change json-ld admin API except for making `verkey` optional on `/jsonld/verify` requests.

### CODE REVIEW

File: aries_cloudagent/messaging/jsonld/create_verify_data.py [hunk 1]
```diff
@@ -38,53 +42,66 @@ def _canonize_document(doc):
...
-
-    return (framed, hash_of_canonized_signature_options + hash_of_canonized_document)
+    if len(data) > len(framed):
+        # > check indicates dropped attrs < is a different error
+        # attempt to collect error report data
+        for_diff = jsonld.compact(expanded, data.get("@context"))
+        dropped = set(data.keys()) - set(for_diff.keys())
+        raise DroppedAttributeError(
+            f"{dropped} attributes dropped. "
+            " Provide definitions in context to correct."
```
Comment (sklump) [inline aries_cloudagent/messaging/jsonld/create_verify_data.py:82]: extra space before Provide.
I know, the most pedentic quibble possible.

File: aries_cloudagent/messaging/jsonld/create_verify_data.py [hunk 2]
```diff
@@ -38,53 +42,66 @@ def _canonize_document(doc):
...
+        if len(data_attribute) > len(frame_attribute):
+            for_diff = jsonld.compact(expanded, data_context)
+            for_diff_attribute = for_diff.get(mapping[1], {})
+            dropped = set(data_attribute.keys()) - set(for_diff_attribute.keys())
+            raise DroppedAttributeError(
+                f"in {mapping[0]}, {dropped} attributes dropped. "
+                "Provide definitions in context to correct."
+            )
+
+    cannonized_signature_options = _canonize_signature_options(signature_options)
```
Comment (sklump) [inline aries_cloudagent/messaging/jsonld/create_verify_data.py:102]: canon not cannon - it's ecclesiastical canon not martial cannon

File: aries_cloudagent/messaging/jsonld/routes.py [hunk 3]
```diff
@@ -3,26 +3,44 @@
...
 
 
 class SignRequestSchema(OpenAPISchema):
     """Request schema for signing a jsonld doc."""
 
-    verkey = fields.Str(required=True, description="Verkey to use for signing")
-    doc = fields.Dict(required=True, description="JSON-LD document to sign")
+    verkey = fields.Str(required=True, description="verkey to use for signing")
+    doc = fields.Nested(
+        Schema.from_dict(
```
Comment (sklump) [inline aries_cloudagent/messaging/jsonld/routes.py:24]: all these years and I never knew about Schema.from_dict()!

File: aries_cloudagent/messaging/jsonld/routes.py [hunk 4]
```diff
@@ -95,40 +128,49 @@ async def verify(request: web.BaseRequest):
...
     """Amend swagger API."""
-
     # Add top-level tags description
     if "tags" not in app._state["swagger_dict"]:
         app._state["swagger_dict"]["tags"] = []
     app._state["swagger_dict"]["tags"].append(
         {
             "name": "jsonld",
-            "description": "JSON-LD message operations",
+            "description": "sign and verify json-ld data.",
```
Comment (sklump) [inline aries_cloudagent/messaging/jsonld/routes.py:173]: Conventionally, throughout aca-py the description starts with a capital but has no trailing period.

File: aries_cloudagent/resolver/did_resolver.py [hunk 5]
```diff
@@ -9,10 +9,15 @@
 from itertools import chain
 from typing import Union
 
-from pydid import DID, DIDUrl, DIDDocument, Service, VerificationMethod
+from pydid import DID, DIDUrl, DIDDocument, Service, VerificationMethod, DIDError
```
Comment (sklump) [inline aries_cloudagent/resolver/did_resolver.py:12]: alphabetical order here helps humans find imports in troubleshooting if need be

**Review by sklump — CHANGES_REQUESTED** [review]
Looks great, just a few tiny quibbles.

File: aries_cloudagent/messaging/jsonld/create_verify_data.py (hunk 2 above)
Comment (burdettadam) [reply aries_cloudagent/messaging/jsonld/create_verify_data.py]: Ahh! make peace, not war! I will correct that.

File: aries_cloudagent/messaging/jsonld/routes.py (hunk 4 above)
Comment (burdettadam) [reply aries_cloudagent/messaging/jsonld/routes.py]: I see. I will go and update the descriptions to follow conventions.

File: aries_cloudagent/resolver/did_resolver.py (hunk 5 above)
Comment (burdettadam) [reply aries_cloudagent/resolver/did_resolver.py]: I agree! I have sorted the relevant imports.

File: aries_cloudagent/messaging/jsonld/routes.py (hunk 3 above)
Comment (burdettadam) [reply aries_cloudagent/messaging/jsonld/routes.py:22]: Schema.from_dict() is nice when it works. I find it convenient to use. Let me know if it goes against conventions.

**Comment by peacekeeper** [comment]
Looks great!

**Review by sklump — APPROVED** [review]

**Pull request merged**